cli/azd/internal/repository/initializer.go

```diff
@@ -343,6 +344,11 @@ func (i *Initializer) InitializeMinimal(ctx context.Context, azdCtx *azdcontext.
 		return err
 	}
 
+	// Check if the project name is valid.
+	if err := names.ValidateProjectName(name); err != nil {
+		return err
+	}
+
```

Comment:
FYI, This is actually dead code here -- `ProjectName()` already cleans the auto-generated name.

@vhvb1989 Just checking from your side, was your recommendation to validate azure.yaml on Load()?

Reply:
on `Load()` might be too late, @weikanglim .  AZD needs to check for valid name when it prompts the user for a project name.  It would be a bad experience to prompt and allow folks to input `foo/bar` just to fail on the next call to azd while trying to load the project again.  

Checking during `load()` could be good for anyone manually creating azure.yaml.  That could be either part of this same PR or as a new PR.  I think the rules about the valid names are already defined in the YAML schema, so I would expect any manual writers to see a red line in the name if they do something like `foo/bar`.  So I am a little less worried about those cases and only currently laser focus on blocking AZD prompt from allowing `foo/bar` as the name of the project

Reply:
Thanks for the explanations!

I did miss the part where we do actually prompt the user for a name here, and softly validating in the yaml schema without a hard break sounds reasonable.

@Menghua1 I agree with @JeffreyCA that a re-prompting should happen -- not an immediate failure. It's an important piece of the first-time experience. Would you be able to address this?

Reply:
> Also wondering if ProjectName() should also strip away leading/trailing hyphens? When the working directory is named something like _app, ProjectName() turns it into -app which is invalid

That's a little odd.. We do have a test case for this: 

```golang
{"StartEndHyphens", "-abc-", "abc"},
```

And it did work for me here:

<img width="867" height="378" alt="Image" src="https://github.com/user-attachments/assets/670cdd95-b298-4e68-9c85-4ba74c541d51" />

Reply:
Ah I had consecutive underscores: `__app` -> `-app` 😅

Reply:
Looks good, thanks!